test/core/tsi/ssl_transport_security_util_test.cc

```diff
@@ -0,0 +1,353 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "src/core/tsi/ssl_transport_security_util.h"
+
+#include <string>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <openssl/bio.h>
+#include <openssl/crypto.h>
+#include <openssl/ssl.h>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+
+#include "src/core/tsi/transport_security.h"
+#include "src/core/tsi/transport_security_interface.h"
+#include "test/core/util/test_config.h"
+
+namespace grpc_core {
+namespace testing {
+
+using ::testing::ContainerEq;
+using ::testing::NotNull;
+using ::testing::Test;
+using ::testing::TestWithParam;
+using ::testing::ValuesIn;
+
+constexpr std::size_t kMaxPlaintextBytesPerTlsRecord = 16384;
+constexpr std::size_t kTls13RecordOverhead = 22;
+
+// Used for debugging.
+int VerifySucceed(X509_STORE_CTX* store_ctx, void* arg) { return 1; }
+
+absl::Status DoHandshake(SSL** out_client, SSL** out_server) {
```

Comment:
I know this might get too long so you want to put it in a separate function. But if this is just used once, let's just put it in the `SetUp`.

Reply:
I think it is more because of this function only does one thing: drive two SSL objects to finish a complete handshake and output those two SSL objects. (Sorry I should've added a comment here). That being said, WDYT of keeping it as a separate function in case we need to call it somewhere else in the future?

Reply:
I think we can always separate the function out later if necessary. Where will we need this in the future? Would it be within the same file, some other files in the test directory, or some files in main source code as well? The answer might vary in different scenarios.
If within the same file, alternatively, how about making it a static function as well?

Reply:
Added `static` keyword.